docs/development/quickstart.rst

```diff
@@ -0,0 +1,365 @@
+.. _development_quickstart:
+
+=========================
+Development Quickstart
+=========================
+
+.. _contributing_environment:
+
+Creating a development environment
+==================================
+
+To make and test code changes and build the documentation locally you will need to
+create a development environment. If you run into problems at any stage do not hesitate
+to `ask for help <https://www.astropy.org/help.html>`_.
+
+Set up GitHub and Git
+---------------------
+
+To contribute to the astropy repository or build from source, you will need to ensure
+that `Git <https://git-scm.com/>`_ is installed and configured on your machine. `GitHub
+has instructions <https://docs.github.com/en/get-started/quickstart/set-up-git>`__ for
+installing and configuring git, and more details and further resources are available in
+the :ref:`contributing.version_control` section.
+
+Install a C compiler if needed
+------------------------------
+
+How to do this will depend on your platform.
+
+**Windows**
+
+You will need `Build Tools for Visual Studio 2022
+<https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2022>`_.
+
+.. note::
+        You DO NOT need to install Visual Studio 2022.
+        You only need "Build Tools for Visual Studio 2022" found by
+        scrolling down to "All downloads" -> "Tools for Visual Studio".
+        In the installer, select the "Desktop development with C++" Workloads.
+
+Alternative options include:
+
+- Install the necessary components on the command line using `vs_BuildTools.exe <https://learn.microsoft.com/en-us/visualstudio/install/use-command-line-parameters-to-install-visual-studio?source=recommendations&view=vs-2022>`_.
+- Use the `WSL <https://learn.microsoft.com/en-us/windows/wsl/install>`_.
+
+**MacOS**
+
+Install the Developer Tools using ``xcode-select --install``.
+
+Further details and related information can be found at
+https://devguide.python.org/setup/#macos.
+
+**Linux**
+
+For Linux-based installations, you won't have to install any additional components.
+
+.. _contributing.forking:
+
+Create a clone of astropy
+-------------------------
+
+If you have not done so already, you will need your own copy of ``astropy`` to
+build it and/or contribute to the source. Astropy is hosted in the `astropy GitHub repository <https://www.github.com/astropy/astropy>`_ and you need to make a clone.
+
+If you *only want to build astropy* but not contribute, you can run the following and then skip to the :ref:`next section<create-isolated-env>`::
```

Comment:
I wonder if we should not omit this too - i.e., keep this the quickstart guide to making a contribution.

Reply:
This came out of linking to this bit (so re-using these instructions) from the Install page, where you might want to only build the docs or run tests. I also didn't want to make more branches here but otherwise the original Install had independent instructions.